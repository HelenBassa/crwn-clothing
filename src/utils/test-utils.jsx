import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };