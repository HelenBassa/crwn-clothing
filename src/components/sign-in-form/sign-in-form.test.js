import { render } from "../../utils/test-utils";
import SignInForm from "./sign-in-form.component";

it("should render SignInForm component", () => {
  const { container } = render(<SignInForm />);
  expect(container).toMatchSnapshot();
});
