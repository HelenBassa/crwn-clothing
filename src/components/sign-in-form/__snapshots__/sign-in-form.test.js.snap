// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignInForm component 1`] = `
<div>
  <div
    class="sc-pyfCe fUYaTT"
  >
    <h2>
      Already have an account?
    </h2>
    <span>
      Sign in with your email and password
    </span>
    <form>
      <div
        class="sc-dkrFOg fSosQj"
      >
        <input
          class="sc-gswNZR hmZPfH"
          name="email"
          required=""
          type="email"
          value=""
        />
        <label
          class="sc-bcXHqe hroYIK"
        >
          Email
        </label>
      </div>
      <div
        class="sc-dkrFOg fSosQj"
      >
        <input
          class="sc-gswNZR hmZPfH"
          name="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="sc-bcXHqe hroYIK"
        >
          Password
        </label>
      </div>
      <div
        class="sc-jrcTuL irwjLS"
      >
        <button
          class="sc-jSUZER cmJkBU"
          type="submit"
        >
          Sign In
        </button>
        <button
          class="sc-jSUZER sc-gKPRtg cmJkBU gAXer"
          type="button"
        >
          Google Sign In
        </button>
      </div>
    </form>
  </div>
</div>
`;
